scaletest/templates/scaletest-runner/scripts/lib.sh

```diff
@@ -247,37 +247,62 @@ set_appearance() {
 		"${CODER_URL}/api/v2/appearance"
 }
 
+namespace() {
+	local ns
+	ns=$(</var/run/secrets/kubernetes.io/serviceaccount/namespace)
+	echo "${ns}"
+}
+coder_pods() {
+	local pods
+	pods=$(kubectl get pods \
+		--namespace "$(namespace)" \
+		--selector "app.kubernetes.io/name=coder,app.kubernetes.io/part-of=coder" \
+		--output jsonpath='{.items[*].metadata.name}')
+	echo "${pods}"
+}
+
 # fetch_coder_full fetches the full (non-slim) coder binary from one of the coder pods
 # running in the same namespace as the current pod.
 fetch_coder_full() {
 	if [[ -x "${SCALETEST_CODER_BINARY}" ]]; then
 		log "Full Coder binary already exists at ${SCALETEST_CODER_BINARY}"
 		return
 	fi
-	local pod
-	local namespace
-	namespace=$(</var/run/secrets/kubernetes.io/serviceaccount/namespace)
-	if [[ -z "${namespace}" ]]; then
+	ns=$(namespace)
+	if [[ -z "${ns}" ]]; then
 		log "Could not determine namespace!"
 		exit 1
 	fi
-	log "Namespace from serviceaccount token is ${namespace}"
-	pod=$(kubectl get pods \
-		--namespace "${namespace}" \
-		--selector "app.kubernetes.io/name=coder,app.kubernetes.io/part-of=coder" \
-		--output jsonpath='{.items[0].metadata.name}')
+	log "Namespace from serviceaccount token is ${ns}"
+	pods=$(coder_pods)
+	if [[ -z ${pods} ]]; then
+		log "Could not find coder pods!"
+		exit 1
+	fi
+	pod=$(echo "${pods}" | cut -d ' ' -f 1)
 	if [[ -z ${pod} ]]; then
 		log "Could not find coder pod!"
 		exit 1
 	fi
 	log "Fetching full Coder binary from ${pod}"
 	# We need --retries due to https://github.com/kubernetes/kubernetes/issues/60140 :(
 	maybedryrun "${DRY_RUN}" kubectl \
-		--namespace "${namespace}" \
+		--namespace "${ns}" \
 		cp \
 		--container coder \
 		--retries 10 \
 		"${pod}:/opt/coder" "${SCALETEST_CODER_BINARY}"
 	maybedryrun "${DRY_RUN}" chmod +x "${SCALETEST_CODER_BINARY}"
 	log "Full Coder binary downloaded to ${SCALETEST_CODER_BINARY}"
 }
+
+# set_status_annotation annotates the currently running pod with the key
+# com.coder.scaletest.status. It will overwrite the previous status.
+set_status_annotation() {
```

Comment:
Maybe we can rename this to `set_pod_status[_annotation]`? Should we also call it automatically from `set_status` so that it's always up-to-date with scale test status? If you need lower case you can do `tr '[A-Z]' '[a-z]' <<<"$1"`.

```sh
❯ tr '[A-Z]' '[a-z]' <<<Preparing
preparing
```

Reply:
That's a good idea!